### PR TITLE
Printed summary on run end

### PR DESCRIPTION
**Pull request by kraenhansen**
This solves #15 by writing summary to the output.
I added a succeeding and pending test to verify the output and I also added a GitHub Actions workflow to test this:

Screenshot 2023-04-06 at 20 14 31

Feel free to pick whatever you want from the branch.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const core = require('@actions/core');
 const { issueCommand } = require('@actions/core/lib/command');
 const mocha = require('mocha');
+const milliseconds = require('ms');
```
Comment (kraenhansen) [inline index.js:4]: Mocha brings this.

File: index.js [hunk 2]
```diff
@@ -22,11 +23,10 @@ function GithubActionsReporter(runner, options) {
 
     const failures = [];
 
-    runner.on(mocha.Runner.constants.EVENT_TEST_FAIL, function(test) {
+    runner.on(mocha.Runner.constants.EVENT_TEST_FAIL, (test) => {
         failures.push(test);
     });
-
-    runner.once(mocha.Runner.constants.EVENT_RUN_END, function() {
+    runner.once(mocha.Runner.constants.EVENT_RUN_END, () => {
```
Comment (kraenhansen) [inline index.js:29]: I needed the body to access `this` from the parent scope.

File: index.js [hunk 3]
```diff
@@ -36,6 +36,14 @@ function GithubActionsReporter(runner, options) {
...
             core.endGroup();
         }
+        core.summary.addRaw(`:white_check_mark: ${this.stats.passes || 0} passing (${milliseconds(this.stats.duration)})`, true);
+        if (this.stats.pending) {
+            core.summary.addRaw(`:pause_button: ${this.stats.pending} pending`, true);
+        }
+        if (this.stats.failures) {
+            core.summary.addRaw(`:x: ${this.stats.failures} failing`, true);
+        }
+        core.summary.write();
```
Comment (kraenhansen) [inline index.js:46]: I've tried to keep this as close to the base epilogue as possible:

https://github.com/mochajs/mocha/blob/37deed262d4bc0788d32c66636495d10038ad398/lib/reporters/base.js#L374-L396

Feel free to adjust or remove the [emojis](https://github.com/ikatyang/emoji-cheat-sheet/blob/master/README.md).

**Comment by daniellockyer** [comment]
Nice, thanks!

**Pull request merged**